SalesforceCommerceCloud/pwa-kit
# @W-14016114: Move Storefront Preview Set up to the SDK

# Description
This PR creates StorefrontPreview component to set up Preview for storefront



# Types of Changes




> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`

# Changes

- (change1)

# How to Test-Drive This PR

- Check out the code
- `STOREFRONT_PREVIEW=true npm run start` at package/pwa-kit-retail-app to turn on Preview feature
- Run local runtime admin at localhost:4000
- Go to `http://localhost:4000/salesforce-systems/scaffold-pwa/test-env-2/preview` to check if storefront is able to communicate with Runtime Admin

// Run storefront with feature
- Re-run storefront with just `npm run start`
- Load the storefront from Runtime Admin: http://localhost:4000/salesforce-systems/scaffold-pwa/test-env-2/preview
- The storefront should not be able to communicate with Runtime Admin



# Checklists




## General


## Accessibility Compliance

You must check off all items in **one** of the follow two lists:


_or..._


## Localization
